csrc/scheduler/matmul.cpp

```diff
@@ -350,50 +367,94 @@ void prologSwizzle(TensorView* shared_mem_tv, const MatmulParams& params) {
      *   7|          |          |
      *    +----------+----------+
      */
-
+    int64_t swizzle_period = n_rows / repeated_pattern_size;
+    // tile_size_y will be splitted by n_cols and then by swizzle_period,
+    // Recalculate swizzle_period if tile_size_y is smaller than the
+    // multiplication of these two split factors. If this happens,
+    // bank conflicts can't be fully removed.
+    if (tile_size_y < n_cols * swizzle_period) {
+      swizzle_period = tile_size_y / n_cols;
+      repeated_pattern_size = n_rows / swizzle_period;
```

Comment:
Why `repeated_pattern_size` changed? Isn't this from mathematical principle and should never change?

Reply:
This is becuase of `swizzle_period = std::gcd(n_rows / repeated_pattern_size, tile_size_y / n_cols)`
If the value of `swizzle_period` is not equal to `n_rows / repeated_pattern_size`, then there will be an error in the swizzle which requires the swizzed 2 dimensions have the same size.
```
    // update repeated_pattern_size to ensure n_rows / repeated_pattern_size
    // equals to swizzle_period otherwise, this is required by 2D swizzle.
    repeated_pattern_size = n_rows / swizzle_period;
```